packages/legend-application-studio/src/components/editor/edit-panel/service-editor/BulkServiceRegistrationEditor.tsx

```diff
@@ -0,0 +1,283 @@
+      );
+    }
+  };
+  const disableRegistration =
+    !selectedEnvOption ||
+    !selectedServiceType ||
+    registrationState.registrationState.isInProgress;
+
+  return (
+    <div
+      data-testid={LEGEND_STUDIO_TEST_ID.SERVICE_REGISTRATION_EDITOR}
+      className="service-registration-editor"
+    >
+      <div className="panel__header">
+        <div className="panel__header__title">
+          <div className="panel__header__title__label">Register Service</div>
+        </div>
+        <div className="panel__header__actions">
+          <div className="panel__header__action">
+            <button
+              className="btn--dark model-loader__header__load-btn"
+              onClick={registerService}
+              disabled={disableRegistration}
+              tabIndex={-1}
+              title="Register Service"
+            >
+              Register
+            </button>
+            <Dialog
+              open={editorStore.bulkServiceRegistrationState.showSuccessModel}
+            >
+              <Modal darkMode={true} className="blocking-alert__summary-text">
+                <div>
+                  <button
+                    className="query-builder__dialog__header__action"
+                    tabIndex={-1}
+                    onClick={() => {
+                      editorStore.bulkServiceRegistrationState.setSuccessModal(
+                        false,
+                      );
+                    }}
+                  >
+                    <TimesIcon />
+                  </button>
+                </div>
+                <ModalBody>
+                  <div>
+                    Successful Services:
+                    {editorStore.bulkServiceRegistrationState.serviceConfigState.registrationResult?.successfulServices.map(
+                      (service) => (
+                        <div key={service}>
+                          <a
+                            href={editorStore.bulkServiceRegistrationState.serviceConfigState.registrationResult?.serviceLinks.at(
+                              editorStore.bulkServiceRegistrationState.serviceConfigState.registrationResult.successfulServices.indexOf(
+                                service,
+                              ),
+                            )}
+                          >
+                            {service}
+                          </a>
+                        </div>
+                      ),
+                    )}
+                  </div>
+                  <>
+                    {editorStore.bulkServiceRegistrationState.serviceConfigState.registrationResult?.successfulServices.forEach(
+                      (element) => {
+                        <div>{element}</div>;
+                      },
+                    )}
+                  </>
+                  <div>
+                    Failed Services:{' '}
+                    {editorStore.bulkServiceRegistrationState.serviceConfigState.registrationResult?.failedServices.join(
```

Comment:
Don't render lists like this. use a list of dom elements, like so:
`items.map(e => <div></div>)`